mne/io/tests/test_pick.py

```diff
@@ -68,7 +68,10 @@ def _channel_type_old(info, idx):
     ch = info['chs'][idx]
 
     # iterate through all defined channel types until we find a match with ch
-    for t, rules in get_channel_types().items():
+    # go in order from most specific (most rules entries) to least specific
+    channel_types = sorted(
+        get_channel_types().items(), key=lambda x: len(x[1]))[::-1]
```

Comment:
can you help me understand this?

Reply:
Sure. One of our rules is that if the `kind` is `CH_EEG` and `coil_type` is `EEG_CSD`, it's a CSD channel. Another rule (the one in `master`) is that if the `kind` is `CH_EEG`, it's an EEG channel.

This sorting by the number of rules sorts by the specificity of the rule, so it looks for CSD before generic EEG.

Another potential solution would be to make the EEG one more specific, to only consider EEG things that are CH_EEG and coil_type EEG. But see for example the changes to `mne/io/ctf/info.py` where previously we did not set it to `EEG` so it was set to NONE. Files converted by previous versions of MNE-Python and any version of MNE-C would have their EEG channels improperly excluded during plotting.

Also we have a `EEG_BIPOLAR` channel type that we need to add if we want to go this more-specific route.

Reply:
I add a comment here to explain this.

Reply:
That is what the existing "go in order from most specific (most rules entries) to least specific" is for...?

Reply:
what specific means in this context was not clear to me.